### PR TITLE
Credentials REST APIs implemented

### PR DESCRIPTION
**Pull request by lorthirk**
Implements #137

### CODE REVIEW

File: rest-api/src/main/java/org/eclipse/kapua/app/api/v1/resources/Credentials.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
+package org.eclipse.kapua.app.api.v1.resources;
```
Comment (kartben) [inline rest-api/src/main/java/org/eclipse/kapua/app/api/v1/resources/Credentials.java]: copyright headers?

File: service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/CredentialXmlRegistry.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package org.eclipse.kapua.service.authentication.credential;
```
Comment (kartben) [inline service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/CredentialXmlRegistry.java]: copyright headers?

File: service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/KapuaExistingCredentialException.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
```
Comment (kartben) [inline service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/KapuaExistingCredentialException.java]: copyright headers?

Comment (lorthirk) [inline service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/KapuaExistingCredentialException.java]: Just below this line. Adding the copyright to the other new files.

File: rest-api/src/main/java/org/eclipse/kapua/app/api/v1/resources/Credentials.java (hunk 1 above)
Comment (lorthirk) [inline rest-api/src/main/java/org/eclipse/kapua/app/api/v1/resources/Credentials.java]: Oops, I forgot them. Fixing in the next commit.

File: service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/CredentialXmlRegistry.java (hunk 2 above)
Comment (lorthirk) [inline service/security/authentication/api/src/main/java/org/eclipse/kapua/service/authentication/credential/CredentialXmlRegistry.java]: Oops, I forgot them. Fixing in the next commit.

**Pull request merged**